patterns/builder.md

```diff
@@ -2,46 +2,58 @@
 
 ## Description
 
-Construct an object with calls to a builder helper. 
-
+Construct an object with calls to a builder helper.
 
 ## Example
 
 ```rust
-struct Foo {
+#[derive(Debug)]
+pub struct Foo {
     // Lots of complicated fields.
+    bar: String,
+}
+
+impl PartialEq for Foo {
+    fn eq(&self, other: &Self) -> bool {
+        self.bar == other.bar
+    }
 }
 
-struct FooBuilder {
+pub struct FooBuilder {
     // Probably lots of optional fields.
-    ...
+    bar: String,
 }
 
 impl FooBuilder {
-    fn new(...) -> FooBuilder {
+    pub fn new() -> FooBuilder {
         // Set the minimally required fields of Foo.
+        FooBuilder {
+            bar: "X".to_string(),
+        }
     }
 
-    fn named(mut self, name: &str) -> FooBuilder {
+    pub fn name(mut self, bar: &str) -> FooBuilder {
         // Set the name on the builder itself, and return the builder by value.
+        self.bar = bar.to_string();
+        self
     }
 
-    // More methods that take `mut self` and return `FooBuilder` setting up
-    // various aspects of a Foo.
-    ...
-
     // If we can get away with not consuming the Builder here, that is an
-    // advantage. It means we can use the builder as a template for constructing
-    // many Foos.
-    fn finish(&self) -> Foo {
+    // advantage. It means we can use the FooBuilder as a template for constructing many Foos.
+    pub fn build(self) -> Foo {
         // Create a Foo from the FooBuilder, applying all settings in FooBuilder to Foo.
+        Foo { bar: self.bar }
     }
 }
 
-fn main() {
-    let f = FooBuilder::new().named("Bar").with_attribute(...).finish();
+#[test]
+fn builder_test() {
+    let foo = Foo {
+        bar: String::from("Y"),
+    };
+    let foo_from_builder: Foo = FooBuilder::new().name("Y").build();
```

Comment:
Do we need this type signature, isn't it inferred?

Reply:
It's not needed but best practice to do so, I think 🤔

Reply:
I am not sure if it is best practice but for docs examples (including rust), most types are explicitly state to let others to understand easily but I think this example should be way to simple, if they get to this page they should already be able to see through this simple example.